### PR TITLE
Ensure breakpoints are in the intended scope by tying them to a print…

### PR DESCRIPTION
**Pull request by adrian-prantl**
… statement.

With the more accurate scopes emitted by https://github.com/apple/swift/pull/65783 this actually matters.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test with https://github.com/apple/swift/pull/65783

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/65783
@swift-ci test

File: lldb/test/API/lang/swift/async/frame/variable/main.swift [hunk 1]
```diff
@@ -7,9 +7,11 @@ func inner<T>(_ t: T) async {
   // is unknown.
   let d = t
   let a = await randInt(30)
-  let b = await randInt(a + 11) // break one
+  let b = await randInt(a + 11)
+  print("break one"
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/frame/variable/main.swift]: missing closing `)`

File: lldb/test/API/lang/swift/stepping/TestSwiftStepping.py [hunk 2]
```diff
@@ -220,6 +220,7 @@ def do_test(self):
         thread.StepOut()
         self.hit_correct_line(thread, "Second case with a where statement")
 
+        thread.StepOver()
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/stepping/TestSwiftStepping.py:223]: why the extra StepOver?

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/65783
@swift-ci test

File: lldb/test/API/lang/swift/stepping/TestSwiftStepping.py (hunk 2 above)
Comment (adrian-prantl) [reply lldb/test/API/lang/swift/stepping/TestSwiftStepping.py:223]: We now go back to the case once. This might be fixable in the compiler.

**Review by kastiglione — APPROVED** [review]

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/65783
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/65783
@swift-ci test

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/65783
@swift-ci test

**Pull request merged**